streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -1323,6 +1327,51 @@ public void shouldDisableMetricCollectionOnMainConsumerOnly() {
         );
     }
 
+    @Test
+    public void shouldThrowConfigExceptionWhenAdminClientMetricsDisabledStreamsMetricsPushEnabled() {
+        props.put(StreamsConfig.adminClientPrefix(AdminClientConfig.ENABLE_METRICS_PUSH_CONFIG), false);
+
+        final Exception exception = assertThrows(ConfigException.class, () -> new StreamsConfig(props));
+
+        assertThat(
+                exception.getMessage(),
+                containsString("KafkaStreams has metrics push enabled" +
+                        " but the admin client metrics push is disabled. Enable " +
+                        " metrics push for the admin client")
+        );
+        assertNull(
```

Comment:
Seems this in unrelated to this test? If we want to verify this, might be best to add new test case?